objcgen/EqualsHelper.cs

```diff
@@ -7,7 +7,7 @@ public class EqualsHelper : MethodHelper{
 		public EqualsHelper (SourceWriter headers, SourceWriter implementation) : base (headers, implementation)
 		{
 			MonoSignature = "Equals(object)";
-			ObjCSignature = "isEqual:(id _Nullable)other";
+			ObjCSignature = "areEqual:(id _Nullable)other";
```

Comment:
The fact that it compiles (the existing tests) is because the default (from `NSObject` type) implementation of `isEqual:` still exists (but it's not the one we want to be used).